x/ccv/provider/keeper/relay.go

```diff
@@ -126,80 +127,86 @@ func (k Keeper) EndBlockVSU(ctx sdk.Context) {
+		// check if CCV channel is establish and send
 		if channelID, found := k.GetChainToChannel(ctx, chainID); found {
-			// Send pending VSC packets to consumer chain
-			k.SendPendingVSCPackets(ctx, chainID, channelID)
-		}
-
-		// check whether there are changes in the validator set;
-		// note that this also entails unbonding operations
-		// w/o changes in the voting power of the validators in the validator set
-		unbondingOps, _ := k.GetUnbondingOpsFromIndex(ctx, chainID, valUpdateID)
-		if len(valUpdates) != 0 || len(unbondingOps) != 0 {
-			// construct validator set change packet data
-			packetData := ccv.NewValidatorSetChangePacketData(valUpdates, valUpdateID, k.ConsumeSlashAcks(ctx, chainID))
-
-			// check whether there is an established CCV channel to this consumer chain
-			if channelID, found := k.GetChainToChannel(ctx, chainID); found {
-				// send this validator set change packet data to the consumer chain
-				err := utils.SendIBCPacket(
-					ctx,
-					k.scopedKeeper,
-					k.channelKeeper,
-					channelID,          // source channel id
-					ccv.ProviderPortID, // source port id
-					packetData.GetBytes(),
-					k.GetParams(ctx).CcvTimeoutPeriod,
-				)
-				if err != nil {
-					panic(fmt.Errorf("packet could not be sent over IBC: %w", err))
-				}
-				// set the VSC send timestamp for this packet
-				k.SetVscSendTimestamp(ctx, chainID, packetData.ValsetUpdateId, ctx.BlockTime())
-			} else {
-				// store the packet data to be sent once the CCV channel is established
-				k.AppendPendingVSC(ctx, chainID, packetData)
-			}
+			k.SendVSCPacketsToChain(ctx, chainID, channelID)
 		}
-		return true // do not stop the iteration
+		return true // continue iterating chains
 	})
-	k.IncrementValidatorSetUpdateId(ctx)
 }
 
-// Sends all pending ValidatorSetChangePackets to the specified chain
-func (k Keeper) SendPendingVSCPackets(ctx sdk.Context, chainID, channelID string) {
-	pendingPackets := k.ConsumePendingVSCs(ctx, chainID)
+// SendVSCPacketsToChain sends all queued ValidatorSetChangePackets to the specified chain
+func (k Keeper) SendVSCPacketsToChain(ctx sdk.Context, chainID, channelID string) {
+	pendingPackets := k.GetPendingVSCs(ctx, chainID)
 	for _, data := range pendingPackets {
-		// send packet over IBC
-		err := utils.SendIBCPacket(
+		// prepare to send the data to the consumer
+		packet, channelCap, err := utils.PrepareIBCPacketSend(
 			ctx,
 			k.scopedKeeper,
 			k.channelKeeper,
 			channelID,          // source channel id
 			ccv.ProviderPortID, // source port id
 			data.GetBytes(),
-			k.GetParams(ctx).CcvTimeoutPeriod,
+			k.GetCCVTimeoutPeriod(ctx),
 		)
 		if err != nil {
+			// something went wrong when preparing the packet
+			panic(fmt.Errorf("packet could not be prepared for IBC send: %w", err))
+		}
+
+		err = k.channelKeeper.SendPacket(ctx, channelCap, packet)
+
+		if err != nil && clienttypes.ErrClientNotActive.Is(err) {
+			// IBC client is expired!
```

Comment:
just to confirm, this sending of the packet will be retried next block? ie. nothing with SendValidatorUpdates changes